### PR TITLE
(PA-2404)-Add debian 10 buster support

### PR DESCRIPTION
**Pull request by loopinu**

### CODE REVIEW

**Comment by loopinu** [comment]
Depends on PR: https://github.com/puppetlabs/beaker-puppet/pull/107

**Comment by loopinu** [comment]
Depends on PR: https://github.com/puppetlabs/puppet-runtime/pull/155

File: configs/platforms/debian-10-amd64.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+platform "debian-10-amd64" do |plat|
+  plat.servicedir "/lib/systemd/system"
+  plat.defaultdir "/etc/default"
+  plat.servicetype "systemd"
+  plat.codename "buster"
+
+  plat.provision_with "export DEBIAN_FRONTEND=noninteractive; apt-get update -qq; apt-get install -qy --no-install-recommends build-essential devscripts make quilt pkg-config debhelper rsync fakeroot cmake"
+  plat.install_build_dependencies_with "DEBIAN_FRONTEND=noninteractive; apt-get install -qy --no-install-recommends "
```
Comment (denis-susan) [inline configs/platforms/debian-10-amd64.rb]: you aren't installing anything here, is this intended?

Comment (gimmyxd) [reply configs/platforms/debian-10-amd64.rb]: this command will be used to install extra packages that are required using `build_requires`

File: configs/platforms/debian-10-amd64.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+platform "debian-10-amd64" do |plat|
+  plat.servicedir "/lib/systemd/system"
+  plat.defaultdir "/etc/default"
+  plat.servicetype "systemd"
+  plat.codename "buster"
+
+  plat.provision_with "export DEBIAN_FRONTEND=noninteractive; apt-get update -qq; apt-get install -qy --no-install-recommends build-essential devscripts make quilt pkg-config debhelper rsync fakeroot cmake"
```
Comment (gimmyxd) [inline configs/platforms/debian-10-amd64.rb]: could you extract the packages in an array, that will make this line shorter

**Comment by gimmyxd** [comment]
@loopinu check travis build, there is a rubocop warning.

**Pull request merged**